src/Middleware/Options/UriReplacementHandlerOption.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Kiota.Abstractions;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+/// <summary>
+/// Interface for making URI replacements.
+/// </summary>
+public interface IUriReplacementHandlerOption : IRequestOption
+{
+    /// <summary>
+    /// Check if URI replacement is enabled for the option.
+    /// </summary>
+    /// <returns>true if replacement is enabled or false otherwise.</returns>
+    bool IsEnabled();
```

Comment:
this could be a getter instead of a method. I don't have a strong opinion on that though

Reply:
This follows the Go format.

Reply:
right, but go doesn't have a notion of accessors within the language (it's all methods).
To be clear what I'm suggesting is `bool IsEnabled { get; set;}` instead